docs/reference/data-streams/tsds.asciidoc

```diff
@@ -0,0 +1,298 @@
+more <<time-series-dimension,dimension fields>>. The matching index template for
+a TSDS must contain mappings for at least one `keyword` dimension.
++
+TSDS documents also typically
+contain one or more <<time-series-metric,metric fields>>.
+
+* {es} generates a hidden <<tsid,`_tsid`>> metadata field for each document in a
+TSDS.
+
+* A TSDS uses <<time-bound-indices,time-bound backing indices>> to store data
+from the same time period in the same backing index.
+
+* The matching index template for a TSDS must contain the `index.routing_path`
+index setting. A TSDS uses this setting to perform
+<<dimension-based-routing,dimension-based routing>>.
+
+* A TSDS uses internal <<index-modules-index-sorting,index sorting>> to order
+shard segments by `_tsid` and `@timestamp`.
+
+* TSDS documents only support auto-generated document `_id` values. For TSDS
+documents, the document `_id` is a hash of the document's dimensions and
+`@timestamp`. A TSDS doesn't support custom document `_id` values.
+
+[discrete]
+[[time-series]]
+=== What is a time series?
+
+A time series is a sequence of observations for a specific entity. Together,
+these observations let you track changes to the entity over time. For example, a
+time series can track:
+
+* CPU and disk usage for a computer
+* The price of a stock
+* Temperature and humidity readings from a weather sensor.
+
+.Time series of weather sensor readings plotted as a graph
+image::images/data-streams/time-series-chart.svg[align="center"]
+
+In a TSDS, each {es} document represents an observation, or data point, in a
+specific time series. Although a TSDS can contain multiple time series, a
+document can only belong to one time series. A time series can't span multiple
+data streams.
+
+[discrete]
+[[time-series-dimension]]
+==== Dimensions
+
+Dimensions are field names and values that, in combination, identify a
+document's time series. Dimensions don't change across documents in the same
+time series. In most cases, a dimension describes some aspect of the entity
+you're measuring. For example, documents related to the same weather sensor may
+always have the same `sensor_id` and `location` values.
+
+You mark a field as a dimension using the boolean `time_series_dimension`
+mapping parameter. The following field types support the `time_series_dimension`
+parameter:
+
+* <<keyword-field-type,`keyword`>>
+* <<ip,`ip`>>
+* <<number,`byte`>>
+* <<number,`short`>>
+* <<number,`integer`>>
+* <<number,`long`>>
+* <<number,`unsigned_long`>>
+
+[[dimension-limits]]
+.Dimension limits
+****
+In a TSDS, {es} uses dimensions to generate the document `_id` and
+<<tsid,`_tsid`>> values. To prevent overly large values, {es} limits the number
+of dimensions for an index using the
+<<index-mapping-dimension-fields-limit,`index.mapping.dimension_fields.limit`>>
+index setting. While you can increase this limit, the resulting document `_id`
+and `_tsid` values can't exceed 32KB.
```

Comment:
`_tsid`'s "native" format is a map of the dimension values - you see if in `fields` or `terms` agg - and any fancy new aggs we will build. But `_id`'s native format is a bunch of letters - *technically* it's a base64 encoded string. That's *technically* made up of some hashes, but in our grand tradition, when we base64 encode things they are "ours" and no one else should try and take them apart. We offer no guarantees to the internals - right now it's two hashes and a long for the timestamp. And it's important that it's influenced by the dimension and timestamp - but there are no rules about how - we allow ourselves the freedom to change it in future versions.

Reply:
I changed this one to:
```
In a TSDS, {es} uses dimensions to
generate the document `_id` and <<tsid,`_tsid`>> values. The resulting `_id` is
always a short encoded hash. To prevent the `_tsid` value from being overly
large, {es} limits the number of dimensions for an index using the
<<index-mapping-dimension-fields-limit,`index.mapping.dimension_fields.limit`>>
index setting. While you can increase this limit, the resulting document `_tsid`
value can't exceed 32KB. 
```